xls/modules/zstd/frame_header_test.cc

```diff
@@ -0,0 +1,379 @@
+    std::cout << "zstd_fh->blockSizeMax: 0x" << fh->blockSizeMax << "\n";
+    std::cout << "zstd_fh->frameType: 0x" << fh->frameType << "\n";
+    std::cout << "zstd_fh->headerSize: 0x" << fh->headerSize << "\n";
+    std::cout << "zstd_fh->dictID: 0x" << fh->dictID << "\n";
+    std::cout << "zstd_fh->checksumFlag: 0x" << fh->checksumFlag << "\n";
+  }
+
+  // Form DSLX Value representing ZSTD Frame header based on data parsed with
+  // ZSTD library. Represents DSLX struct `FrameHeader`.
+  Value CreateExpectedFrameHeader(ZSTD_frameHeader* fh,
+                                  FrameHeaderStatus expected_status) {
+    if (expected_status == FrameHeaderStatus::CORRUPTED ||
+        expected_status == FrameHeaderStatus::UNSUPPORTED_WINDOW_SIZE) {
+      return Value::Tuple({
+          /*window_size:*/ Value(UBits(0, 64)),
+          /*frame_content_size:*/ Value(UBits(0, 64)),
+          /*dictionary_id:*/ Value(UBits(0, 32)),
+          /*content_checksum_flag: */ Value(UBits(0, 1)),
+      });
+    }
+    return Value::Tuple({
+        /*window_size:*/ Value(UBits(fh->windowSize, 64)),
+        /*frame_content_size:*/ Value(UBits(fh->frameContentSize, 64)),
+        /*dictionary_id:*/ Value(UBits(fh->dictID, 32)),
+        /*content_checksum_flag: */ Value(UBits(fh->checksumFlag, 1)),
+    });
+  }
+
+  // Create DSLX Value representing Buffer contents after parsing frame header
+  // in simulation. Represents DSLX struct `Buffer`.
+  Value CreateExpectedBuffer(Value dslx_simulation_input,
+                             absl::Span<const uint8_t> input_buffer,
+                             uint32_t consumed_bytes_count,
+                             FrameHeaderStatus expected_status) {
+    // Return original buffer contents
+    if (expected_status == FrameHeaderStatus::NO_ENOUGH_DATA) {
+      return dslx_simulation_input;
+    }
+    // Critical failure - return empty buffer
+    if (expected_status == FrameHeaderStatus::CORRUPTED ||
+        expected_status == FrameHeaderStatus::UNSUPPORTED_WINDOW_SIZE) {
+      return Value::Tuple({/*contents:*/ Value(UBits(0, dslx_buffer_size)),
+                           /*length:*/ Value(UBits(0, 32))});
+    }
+
+    // Frame Header parsing succeeded. Expect output buffer contents with
+    // removed first `consumed_bytes_count` bytes and extended to
+    // dslx_buffer_size if necessary
+    int bytes_to_extend =
+        dslx_buffer_size_bytes - (input_buffer.size() - consumed_bytes_count);
+    std::vector output_buffer(input_buffer.begin() + consumed_bytes_count,
+                              input_buffer.end());
+    for (int i = 0; i < bytes_to_extend; i++) {
+      output_buffer.push_back(0);
+    }
+
+    auto expected_buffer_contents =
+        Value(Bits::FromBytes(output_buffer, dslx_buffer_size));
+    auto output_buffer_size_bits =
+        (input_buffer.size() - consumed_bytes_count) * CHAR_BIT;
+    uint32_t expected_buffer_size = output_buffer_size_bits > dslx_buffer_size
```

Comment:
should we use `size_t` here or make the conversion explicit?

Reply:
changed to `size_t`